### DISCUSSION TITLE
How to properly achieve different example(s) for same type/model per endpoint?

### DISCUSSION THREAD

**Discussion by jeremyschiemann** [fastapi/fastapi#11170] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Path
from typing import Annotated
from copy import deepcopy

app = FastAPI()

item_id_type = Annotated[str, Path(description='This is my Item ID', ge=10, le=50, example=12)]
item_id_type_b = deepcopy(item_id_type)
item_id_type_b.__metadata__[0].example = 17


@app.get('/route_a')
def operation_a(item_id: item_id_type) -> str
  if item_id == 17:
    raise ValueError('item with id 17 doesnt support this operation')
  return f'id: {item_id}'

@app.get('/route_a')
def operation_b(item_id: item_id_type_b) -> str:
  if item_id == 12:
    raise ValueError('item with id 12 doesnt support this operation')
  return f'id: {item_id}
```

### Description

I'm trying to have a single type that represents an item ID but make different endpoints use a different example.

rightnow I'm using a workaround by copying the original type and then mutate the example like so:
```python
item_id_type_b = deepcopy(item_id_type)
item_id_type_b.__metadata__[0].example = 17
```

Is there a better way to have a single type/model with its validations etc but use different examples per endpoint?
The only other option I see is copy/pasting the Path-definition resulting in a lot of code duplication.

Everything else i tied either didn't work or overrides the Path-definition entirely.

### Operating System

Other

### Operating System Details

WSL (ubuntu) on Windows 10

### FastAPI Version

0.109.2 

### Pydantic Version

2.5.3

### Python Version

3.11.4

**Comment by YuriiMotov** [comment, marked as answer]
This seems to be working:

```
from typing import Annotated

from fastapi import FastAPI, Path

app = FastAPI()

item_id_type = Annotated[
    str,
    Path(description="This is my Item ID", ge=10, le=50, example=1),
]


@app.get("/route_a")
def operation_a(item_id: Annotated[item_id_type, Path(example=12)]) -> str:
    if item_id == 17:
        raise ValueError("item with id 17 doesnt support this operation")
    return f"id: {item_id}"


@app.get("/route_b")
def operation_b(item_id: Annotated[item_id_type, Path(example=17)]) -> str:
    if item_id == 12:
        raise ValueError("item with id 12 doesnt support this operation")
    return f"id: {item_id}"
```

BTW, `ge` and `le` are not working for `str` type fields. And you need to include `{item_id}` to the paths:

```
from typing import Annotated

from fastapi import FastAPI, Path

app = FastAPI()

item_id_type = Annotated[
    int,
    Path(description="This is my Item ID", ge=10, le=50, example=1),
]


@app.get("/route_a/{item_id}")
def operation_a(item_id: Annotated[item_id_type, Path(example=12)]) -> str:
    if item_id == 17:
        raise ValueError("item with id 17 doesnt support this operation")
    return f"id: {item_id}"


@app.get("/route_b/{item_id}")
def operation_b(item_id: Annotated[item_id_type, Path(example=17)]) -> str:
    if item_id == 12:
        raise ValueError("item with id 12 doesnt support this operation")
    return f"id: {item_id}"
```
